backend/tests/services/kanban/Jira/JiraVerifyToken.test.ts

```diff
@@ -216,8 +196,187 @@ describe("verify token and return columns and users", async () => {
+          tokenVerifyModel.site
+        }.atlassian.net/rest/agile/1.0/board/2/issue?maxResults=100&jql=status in ('${doneColumn.join(
+          "','"
+        )}') AND statusCategoryChangedDate >= ${
+          tokenVerifyModel.startTime
+        } AND statusCategoryChangedDate <= ${tokenVerifyModel.endTime}`,
+        {
+          headers: { Authorization: `${tokenVerifyModel.token}` },
+        }
+      )
+      .reply(200, JiraCards);
+
+    const response = await jiraVerifyTokenProto.getAllDoneCards(
+      tokenVerifyModel,
+      doneColumn
+    );
+
+    expect(response.length).deep.equal(1);
+  });
+});
+
+describe("make page query", () => {
+  it("should return cards issues ", async () => {
+    const total = 123;
+    const jql = `status in ('${doneColumn.join(
+      "','"
+    )}') AND statusCategoryChangedDate >= ${
+      tokenVerifyModel.startTime
+    } AND statusCategoryChangedDate <= ${tokenVerifyModel.endTime}`;
+    const cards: any = [];
+    jiraVerifyTokenProto.queryCount = 100;
+    mock
+      .onGet(
+        `https://${
+          tokenVerifyModel.site
+        }.atlassian.net/rest/agile/1.0/board/2/issue?maxResults=100&startAt=100&jql=status in ('${doneColumn.join(
+          "','"
+        )}') AND statusCategoryChangedDate >= ${
+          tokenVerifyModel.startTime
+        } AND statusCategoryChangedDate <= ${tokenVerifyModel.endTime}`,
+        {
+          headers: { Authorization: `${tokenVerifyModel.token}` },
+        }
+      )
+      .reply(200, JiraCards);
+    await jiraVerifyTokenProto.pageQuerying(
+      tokenVerifyModel,
+      total,
+      jql,
+      cards
+    );
+    expect(cards.length).equal(1);
+  });
+});
+
+describe("get cycle time and assignee set", () => {
+  afterEach(() => sinon.restore());
+
+  it("should return cycle time and assigneeset", async () => {
+    const jiraCardKey = "ADM-50";
+    mock
+      .onGet(
+        `https://${tokenVerifyModel.site}.atlassian.net/rest/internal/2/issue/${jiraCardKey}/activityfeed
+      )}')`,
+        {
+          headers: { Authorization: `${tokenVerifyModel.token}` },
+        }
+      )
+      .reply(200, JiraCardCycleTime);
+
+    const response = await JiraVerifyToken.getCycleTimeAndAssigneeSet(
+      jiraCardKey,
+      tokenVerifyModel.token,
+      tokenVerifyModel.site
+    );
+    expect(response.values.length).equal(0);
+  });
+});
+
+describe("query users by cards", () => {
+  afterEach(() => sinon.restore());
+  it("should throw error when allDoneCards length is Zero", async () => {
+    sinon.stub(jiraVerifyTokenProto, "getAllDoneCards").returns("");
+
+    try {
+      await jiraVerifyTokenProto.queryUsersByCards(
+        tokenVerifyModel,
+        doneColumn
+      );
+    } catch (error) {
+      if (error instanceof NoCardsInDoneColumnError) {
+        expect(error.message).equals(
+          "unsupported type: There is no cards in done column."
+        );
+      }
+    }
+  });
+
+  it("should return users", async () => {
```

Comment:
test describe should start with `should .....when(optional)...given(optional)`, so  give dev a result directly